### PR TITLE
[WIP]Add append and remove methods to config transaction family

### PR DESCRIPTION
**Pull request by h4xr**
The PR adds an append and remove method to config blockchain family.
_append_setting_entry allows us to add new values to an existing key
_remove_setting_entry allows us to remove a provided value from an existing key if the provided value exists in the key.

JIRA Issue: STL-261

### CODE REVIEW

File: families/config/sawtooth_config/processor/handler.py [hunk 1]
```diff
@@ -246,18 +246,54 @@ def _get_config_value(state, key, default_value=None):
...
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to save config value {}'.format(key))
+    if setting != 'sawtooth.config.vote.proposals':
+        LOGGER.info('Config setting %s changed from %s to %s',
+                    key, old_value, value)
+
+
+def _append_setting_entry(state, key, value):
```
Comment (jjason) [inline families/config/sawtooth_config/processor/handler.py]: I don't see where this ever gets called.  Is that another story?

File: families/config/sawtooth_config/processor/handler.py [hunk 2]
```diff
@@ -246,18 +246,54 @@ def _get_config_value(state, key, default_value=None):
...
+    address = _make_config_key(key)
+    setting = _get_setting_entry(state, address)
+
+    old_value, old_entry_index = _get_old_entry(setting, key)
     if old_entry_index is not None:
+        value = value + "," + old_value
         setting.entries[old_entry_index].value = value
     else:
         setting.entries.add(key=key, value=value)
```
Comment (jjason) [inline families/config/sawtooth_config/processor/handler.py]: Besides, prepending `new_value,` to the existing value if the key already exists, this function looks the same as `_set_config_value`.  Would recommend practicing DRY and refactoring common code.

File: families/config/sawtooth_config/processor/handler.py [hunk 3]
```diff
@@ -318,6 +318,41 @@ def _append_setting_entry(state, key, value):
                     key, old_value, value)
 
 
+def _remove_setting_entry(state, key, value):
```
Comment (jjason) [inline families/config/sawtooth_config/processor/handler.py]: I don't see this being called anywhere.   Did I missing something or is that for a future story?

Comment (jjason) [inline families/config/sawtooth_config/processor/handler.py]: This function removes a _value_ from a setting.  It doesn't actually remove the setting.  Should it remove the setting if the only or last value is removed or should this function continue doing what it is doing (i.e., when removing the last value the setting is an empty string) and then a new remove setting is created?

File: families/config/sawtooth_config/processor/handler.py [hunk 4]
```diff
@@ -318,6 +318,41 @@ def _append_setting_entry(state, key, value):
...
+
+    if len(addresses) != 1:
+        LOGGER.warning(
+            'Failed to save value on address %s', address)
+        raise InternalError(
+            'Unable to save config value {}'.format(key))
+    if setting != 'sawtooth.config.vote.proposals':
+        LOGGER.info('Value %s removed from key %s',
+                    value, key)
+
```
Comment (jjason) [inline families/config/sawtooth_config/processor/handler.py]: This function also seems to share code with `_set_config_value` and `_append_setting_entry`.  Better to refactor and share common code than repeating it.

File: families/config/sawtooth_config/processor/handler.py (hunk 3 above)
Comment (h4xr) [inline families/config/sawtooth_config/processor/handler.py]: jjason, yes, currently, these are just added to implement the functionality. Still have to add the calls to the functions.

Comment (h4xr) [inline families/config/sawtooth_config/processor/handler.py]: Thanks, will modify this one to remove the setting if it's empty

**Comment by jsmitchell** [comment]
hi @h4xr - are you planning on continuing with this PR? Otherwise I will probably mark it closed.

**Comment by h4xr** [comment]
Hello,
Sorry for a long inactivity, got occupied with some work. Will be working on this PR this weekend. Hoping to complete it by Monday :)

**Comment by h4xr** [comment]
Hi, I am unable to get some free time to complete the PR and hence closing this PR for now. If anyone will like to work on this one, please feel free to do so.
Hope to work along in future with the awesome team.

**Pull request closed without merging**